Pidgin/Parser.Not.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Generic;
+using System.Linq;
+using Pidgin.ParseStates;
+
+namespace Pidgin
+{
+    public static partial class Parser
+    {
+        /// <summary>
+        /// Creates a parser which attempts to apply the specified parser and only succeeds if that parser fails.
+        /// </summary>
+        /// <param name="parser">The parser to apply if the current parser fails without consuming any input</param>
+        /// <returns>A parser which tries to apply the current parser, and applies <paramref name="parser"/> if the current parser fails without consuming any input.</returns>
+        public static Parser<TToken, TToken> Not<TToken>(Parser<TToken, TToken> parser) {
+            return new NegatedParser<TToken>(parser);
+        }
+        
+        private sealed class NegatedParser<TToken> : Parser<TToken, TToken>
+        {
+            private readonly Parser<TToken, TToken> _parser;
+
+            public NegatedParser(Parser<TToken, TToken> parser) : base(parser.Expected)
```

Comment:
The `base` call here says that the `Not` parser expects whatever the wrapped parser expects, which ain't right in this instance. Since we can't know statically what this parser does expect (only what it doesn't), I'd go with `ExpectedUtil<TToken>.Empty`.